src/main/scala/lms/thirdparty/cuda/cuda.scala

```diff
@@ -943,6 +943,26 @@ trait CudaOps extends Dsl with StackArrayOps with SizeTOps with CLibs with CudaF
   }
 }
 
+trait CudaLibs extends CudaOps {
```

Comment:
I am wondering if it is a good idea to move Cuda Kernel implementations to this separate trait called CudaLibs?

Reply:
yeah, this makes sense. I'll do the same for softmax, softmaxGrad after we merge this PR